site/_shortcodes/Hreflang.js

```diff
@@ -0,0 +1,33 @@
+const {findByUrl} = require('../_data/lib/find');
+const path = require('path');
+
+/**
+ * Outputs a list of <link hreflang=""> tags with alternate language versions
+ * for the given url.
+ * @param {string} url Current page url
+ * @param {object} site Site object from _data/site.json
+ * @param {EleventyCollectionObject} collections Eleventy collections object
+ */
+function Hreflang(url, site, collections) {
+  if (!url) {
+    return;
+  }
+  const urlParts = url.split('/');
+  const replace = site.locales.includes(urlParts[1]) ? 1 : 0;
+  const hreflangs = site.locales.reduce((out, locale) => {
+    urlParts.splice(1, replace, locale);
```

Comment:
I don't think this is right. If the source URL was not localized (and `urlParts[1]` is not a language code), then `replace` will always be zero, so we'll _add_ a new language code for each iteration of this `reduce`.

I kind of forget whether _all_ URLs will have a language code at "/<lang>": if so, then the 1/0 check isn't needed, and your code is odd but will always work correctly.

I think ideally you'd strip the previous language code first, outside the loop, then slice the array every iteration and insert something new into it.

Reply:
You're right, the replace bit should be inside the loop. I moved this logic to a separate helper for better readability, and added a test, just to be sure.